platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/moduleconverter/DefaultRootComponentStateBuilder.java

```diff
@@ -41,10 +41,9 @@
 import org.gradle.internal.model.ModelContainer;
 
 import javax.annotation.Nullable;
-import javax.inject.Inject;
 import java.lang.ref.SoftReference;
 
-public class DefaultRootComponentMetadataBuilder implements RootComponentMetadataBuilder, HoldsProjectState {
+public class DefaultRootComponentStateBuilder implements RootComponentStateBuilder, HoldsProjectState {
```

Comment:
This class is still not 100% correct. It still gives the detached configuration's root component the ComponentIdentifier of the project it came from, even though it is a different component.